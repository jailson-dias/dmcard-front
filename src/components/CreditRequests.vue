<template>
  <v-content>
    <div class="d-flex flex-column align-center">
      <h2 class="display-1 mt-9 mb-6">{{requestCreditCardScreenTitle}}</h2>
      <v-list color="transparent">
        <RequestInfoCard
          v-for="(credit, i) in requestCreditCard"
          :key="i + 'request'"
          :score="credit.score"
          :name="credit.name"
          :cpf="credit.cpf"
          :email="credit.email"
          :phone="credit.phone"
          :income="credit.income"
          :credit="credit.credit"
          :id="credit.id"
        />
      </v-list>
    </div>
  </v-content>
</template>

<script>
import RequestInfoCard from "./RequestInfoCard";

import { mapState } from "vuex";

export default {
  components: {
    RequestInfoCard
  },
  computed: {
    ...mapState({
      requestCreditCard: state => state.requestCreditCard,
      requestCreditCardScreenTitle: state => state.requestCreditCardScreenTitle
    })
  }
};
</script>

<style>
</style>