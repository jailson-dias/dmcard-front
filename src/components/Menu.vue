<template>
  <v-navigation-drawer app color="#023D86" dark mobile-break-point="900">
    <div class="d-flex justify-center mt-9 mb-4">
      <v-img src="https://picsum.photos/80/80?random" height="80" width="80" contain></v-img>
    </div>
    <v-list>
      <v-list-item>
        <v-list-item-content>
          <v-list-item-title class="font-weight-bold">Solicitações</v-list-item-title>
        </v-list-item-content>
      </v-list-item>
      <v-divider></v-divider>
      <v-list-item v-for="item in items" :key="item.title" link>
        <v-list-item-content>
          <v-list-item-title class="ml-7">{{ item.title }}</v-list-item-title>
        </v-list-item-content>
      </v-list-item>
    </v-list>

    <template v-slot:append>
      <div class="pa-2">
        <v-btn block>Logout</v-btn>
      </div>
    </template>
  </v-navigation-drawer>
</template>

<script>
export default {
  data() {
    return {
      items: [
        { title: "Dashboard", icon: "dashboard" },
        { title: "Account", icon: "account_box" },
        { title: "Admin", icon: "gavel" }
      ],
      height: 0
    };
  },
  mounted() {
    this.onResize();
  },

  methods: {
    onResize() {
      this.height = window.innerHeight;
    }
  }
};
</script>

<style>
</style>