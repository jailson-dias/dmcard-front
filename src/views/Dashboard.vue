<template>
  <v-container class="pa-0 mx-0">
    <div class="d-flex">
      <Menu />
      <CreditRequests />
    </div>
  </v-container>
</template>

<script>
import Menu from "../components/Menu";
import CreditRequests from "../components/CreditRequests";

export default {
  components: {
    Menu,
    CreditRequests
  }
};
</script>

<style scoped>
</style>