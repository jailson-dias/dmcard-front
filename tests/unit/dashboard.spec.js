import { shallowMount, createLocalVue } from "@vue/test-utils";
import Dashboard from "@/views/Dashboard.vue";
import Menu from "@/components/Menu.vue";
import CreditRequests from "@/components/CreditRequests.vue";
import Router from "vue-router";

const localVue = createLocalVue();
localVue.use(Router);

describe("App page", () => {
  const build = () => {
    const wrapper = shallowMount(Dashboard, { localVue });

    return {
      wrapper,
      Menu: () => wrapper.find(Menu),
      CreditRequests: () => wrapper.find(CreditRequests)
    };
  };

  it("É uma página do vue", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("A página tem o carousel e o form para preencher os dados", () => {
    const { Menu, CreditRequests } = build();

    expect(Menu().exists()).toBe(true);
    expect(CreditRequests().exists()).toBe(true);
  });
});
