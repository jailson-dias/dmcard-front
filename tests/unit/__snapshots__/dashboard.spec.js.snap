// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page É uma página do vue 1`] = `
<div>
  <menu-stub></menu-stub>
  <creditrequests-stub></creditrequests-stub>
</div>
`;
